homeassistant/components/nmbs/sensor.py

```diff
@@ -146,6 +146,10 @@ def device_state_attributes(self):
     def update(self):
         """Set the state equal to the next departure."""
         liveboard = self._api_client.get_liveboard(self._station)
+
+        if liveboard is None or not liveboard["departures"]:
+            return self._state
```

Comment:
Nothing is checking this return value.
```suggestion
            return
```